runtime/include/tasks/qthreads/tasks-qthreads.h

```diff
@@ -118,33 +118,24 @@ volatile int chpl_qthread_done_initializing;
 
 typedef struct {
   chpl_taskID_t id;
+  // lock_filename, lock_lineno
 } chpl_task_bundleData_t;
 
-typedef struct {
-    int32_t task_filename;
-    int task_lineno;
-    chpl_taskID_t id;
-    chpl_bool is_executeOn;
-    c_sublocid_t requestedSubloc;  // requested sublocal for task
-    chpl_task_prvData_t prvdata;
-} chpl_task_prvDataImpl_t;
-
-// Define PRV_DATA_IMPL_VAL to set up a chpl_task_prvData_t.
-#define PRV_DATA_IMPL_VAL(_fn, _ln, _id, _is_execOn, _subloc, _serial) \
-        { .task_filename = _fn, \
-          .task_lineno = _ln, \
-          .id = _id, \
-          .is_executeOn = _is_execOn, \
-          .requestedSubloc = _subloc, \
-          .prvdata = { .serial_state = _serial } }
+// Now that chpl_task_bundleData_t is defined, include
+// chpl-tasks-bundle.h so the full structure is available
+// in the rest of this file.
+#include "chpl-tasks-bundle.h"
 
 // Structure of task-local storage
 typedef struct chpl_qthread_tls_s {
-    /* Task private data: serial state, etc. */
-    chpl_task_prvDataImpl_t chpl_data;
-    /* Reports */
-    int32_t    lock_filename;
-    size_t      lock_lineno;
+  chpl_task_bundle_p bundle;
+  // The below fields could move to chpl_task_bundleData_t
+  // That would reduce the size of the task local storage,
+  // but increase the size of executeOn bundles.
```

Comment:
I'm failing to understand the need for chpl_qthread_tls_t at all.  Why can we not say that qthread_get_tasklocal() returns the address of a chpl_task_bundle_t*, rather than that it returns the address of a chpl_qthread_tls_t struct that contains as its only member a chpl_task_bundle_t*? What does the chpl_qthread_tls_t container get us?